rcl/src/rcl/rmw_implementation_identifier_check.c

```diff
@@ -150,11 +151,17 @@ INITIALIZER(initialize) {
         actual_rmw_impl_id,
         RCL_RET_MISMATCHED_RMW_ID
       );
-      exit(RCL_RET_MISMATCHED_RMW_ID);
+      return RCL_RET_MISMATCHED_RMW_ID;
     }
     // Free the memory now that all checking has passed.
     allocator.deallocate((char *)expected_rmw_impl, allocator.state);
   }
+  return RCL_RET_OK;
+}
+
+INITIALIZER(initialize) {
+  rcl_ret_t check_status = _internal_rmw_check();
```

Comment:
@Blast545 nit: 
```suggestion
  rcl_ret_t ret = _internal_rmw_check();
```
for the sake of consistency with the rest of `rcl`.